### PR TITLE
make introspection.RootDiskType int64 rather than int (#1634)

### PR DESCRIPTION
**Pull request by mwhudson**
Fixes tests (and functionality I suspect!) on 32-bit systems.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,3 +11,5 @@ require (
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
 	gopkg.in/yaml.v2 v2.2.7
 )
+
+go 1.13
```
Comment (jtopjian) [inline go.mod:15]: Can this be removed? It's causing the Travis failure. I'll clean this up in another PR.

**Comment by jtopjian** [comment]
@mwhudson Thank you for looking into this - it's really appreciated.

From your comments in #1634:

> I don't know if it would be too much of an abi break, but 

Breaking changes are OK as long as they are justified.


> github.com/gophercloud/gophercloud/openstack/baremetalintrospection/v1/introspection.RootDiskType being an "int" vs an "int64" definitely seems like a mistake. 

Yes, from the error message given in #1634, I agree.  I'd also be inclined to change other `int`'s to `int64` while we're at it.

**Comment by mwhudson** [comment]
> @mwhudson Thank you for looking into this - it's really appreciated.

No worries, thanks for the quick review!

> From your comments in #1634:
> 
> > I don't know if it would be too much of an abi break, but
> 
> Breaking changes are OK as long as they are justified.

Fair enough.

> > github.com/gophercloud/gophercloud/openstack/baremetalintrospection/v1/introspection.RootDiskType being an "int" vs an "int64" definitely seems like a mistake.
> 
> Yes, from the error message given in #1634, I agree. I'd also be inclined to change other `int`'s to `int64` in this package while we're at it.

All of them? I can do that easily enough, of course. I did have a quick scan and didn't think any of the others would be problematic but it might be best to be consistent.

**Comment by jtopjian** [comment]
> All of them? I can do that easily enough, of course. I did have a quick scan and didn't think any of the others would be problematic but it might be best to be consistent.

Sorry - I may have been too broad in my last message 🙂 

I should have said that we should look at changing them where it makes sense, and only in this introspection package. For example, the Memory fields _might_ work better as an int64. I know it's a long shot to have that much memory, but who knows 2-3 years from now. Things like that.

File: go.mod (hunk 1 above)
Comment (mwhudson) [reply go.mod]: Oh yes, I meant to remove that before pushing, cleaned up now.

**Comment by mwhudson** [comment]
Well OK. Want to merge this, or fix it yourself how ever you like? :)

**Pull request merged**